packages/components/src/border-control/border-control-dropdown/component.tsx

```diff
@@ -171,21 +185,26 @@ const BorderControlDropdown = (
 	const renderToggle: DropdownComponentProps[ 'renderToggle' ] = ( {
 		onToggle,
 	} ) => (
-		<Button
-			onClick={ onToggle }
-			variant="tertiary"
-			aria-label={ toggleAriaLabel }
-			tooltipPosition={ dropdownPosition }
-			label={ __( 'Border color and style picker' ) }
-			showTooltip={ true }
-		>
-			<span className={ indicatorWrapperClassName }>
-				<ColorIndicator
-					className={ indicatorClassName }
-					colorValue={ color }
-				/>
-			</span>
-		</Button>
+		<>
+			<div id={ descriptionId } style={ { display: 'none' } }>
```

Comment:
Hiding descriptions makes them inaccessible.

```suggestion
			<VisuallyHidden id={ descriptionId }>
```

Reply:
In this case, since descriptions are attached to the button via `aria-describedBy`, they get announced when focus is on the related button. Using `<VisuallyHidden/>` it was announced twice: once before the button, and once while focus was on the button.

I was going off of [@afercia's comment](https://github.com/WordPress/gutenberg/pull/54439#issuecomment-1719006321):

> Add an aria-describedby attribute that points to an element containing the additional information. This element can also be hidden with display: none, screen readers will announce the description anyways.

I feel like one announcement semantically linked to the button is better than it announced before the button and again with the button, but I'm also aware I'm not a pro screen reader user so I'll defer to your judgement :)

Reply:
@jeryj Yeah, I tested it and `display: none;` is not allowing the updated description to be read. It just has the default not selected message even after selecting a color.

Should probably place the description outside of the button though.

Reply:
I'll go ahead and change it to `VisuallyHidden`. Which screen reader + browser are you testing on? I'm curious to see if I can use `display: none;` and still get it to update the message after selection so it's not as verbose.

Reply:
@jeryj NVDA with Firefox.

Reply:
Late to the party. Thinking we should really define a best practice for elements targeted by aria-describedby. Technically they can be hidden. Even with a HTML `hidden` attribute. I'd tend to think that if a screen readers don't get the descirption when it updates, it's either a screen reader bug or they just don't expect the description updates, regardles if it's hidden or not.
This should be tested a bit more in depth, with all major browser / screen reader combinations.

Reply:
I agree, @afercia. I'd really rather not have the duplicate announcement. I feel like there has to be a way, and if we can get it figured out, it can become the pattern we use for all aria-describedby.

Reply:
@jeryj @afercia I just tested this again and it seems to be working as is. I am not sure if I hit a bug with gutenberg.run or what the problem actually was. This time, I manually changed the hex code of the color. I wonder if selecting from one of the defined options just doesn't update this value. I think that's all I did before.

Anyway, I think we can for sure do away with `VisuallyHidden` globally for descriptions. Had no idea something like this would work.

I also created a local test.html file and cross checked it in Chrome as well as Firefox. This seems solid.

Reply:
> Anyway, I think we can for sure do away with VisuallyHidden globally for descriptions. Had no idea something like this would work.

Yay! I'll put in some PRs to update it to this method.

Reply:
To be clear, did you test with the current version of this PR that is using `<VisuallyHidden/>`? Or did you edit it to go back to the `display: none` method? I'll update this PR to the `display: none;` method.

Reply:
@jeryj I just tested the current version of the PR but I created a similar situation in a local testing file. If you revert back, will be interested to see if it works.

Reply:
Updated the PR to use `display: none;`. If it _doesn't_ work, then one possibility is that it doesn't work reliably after a selection due to the description div being removed and remounting when the component updates?

Reply:
@jeryj I think it might be the re-mounting issue. It no longer works.

Reply:
I opened an issue to discuss and work on this: https://github.com/WordPress/gutenberg/issues/55362